### PR TITLE
[DCJ-62] Add or remove all DAAs from user

### PR DESCRIPTION
**Pull request by fboulnois**
### Addresses

https://broadworkbench.atlassian.net/browse/DCJ-62

### Summary

Add or remove all DAAs from a user. Requires https://github.com/DataBiosphere/consent/pull/2308 to go in first.

----
Have you read [CONTRIBUTING.md](../CONTRIBUTING.md) lately? If not, do that first.

- Label PR with a Jira ticket number and include a link to the ticket
- Label PR with a security risk modifier [no, low, medium, high]
- PR describes scope of changes
- Get a minimum of one thumbs worth of review, preferably two if enough team members are available
- Get PO sign-off for all non-trivial UI or workflow changes
- Verify all tests go green
- Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

**Review by rushtong — APPROVED** [review]
Looks good, thank you 👍🏽

File: src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java [hunk 1]
```diff
@@ -278,4 +278,52 @@ public Response bulkRemoveUsersFromDaa(
       return createExceptionResponse(e);
     }
   }
+
+  @POST
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, CHAIRPERSON})
```
Comment (aarohinadkarni) [inline src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java]: Same as the other PR, should this be Admin & Signing Official?

File: src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java [hunk 2]
```diff
@@ -278,4 +278,52 @@ public Response bulkRemoveUsersFromDaa(
...
+      }
+      return Response.ok().build();
+    } catch (Exception e) {
+      return createExceptionResponse(e);
+    }
+  }
+
+  @DELETE
+  @Consumes(MediaType.APPLICATION_JSON)
+  @RolesAllowed({ADMIN, CHAIRPERSON})
```
Comment (aarohinadkarni) [inline src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java]: Same as the other PR, should this be Admin & Signing Official?

File: src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java [hunk 3]
```diff
@@ -278,4 +278,52 @@ public Response bulkRemoveUsersFromDaa(
...
+      String json) {
+    try {
+      User authedUser = userService.findUserByEmail(authUser.getEmail());
+      User user = userService.findUserById(userId);
+      if (authedUser.hasUserRole(UserRoles.SIGNINGOFFICIAL) && !Objects.equals(authedUser.getInstitutionId(), user.getInstitutionId())) {
+        return Response.status(Status.FORBIDDEN).build();
+      }
+      List<DataAccessAgreement> daaList = daaService.findDAAsInJsonArray(json, "daaList");
+      for (DataAccessAgreement daa : daaList) {
+        libraryCardService.addDaaToUserLibraryCardByInstitution(user, authedUser.getInstitutionId(), daa.getDaaId());
```
Comment (aarohinadkarni) [inline src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java:298]: Through local testing, I found that when trying to add a DAA that was already on the user we get a 409 in response, but no helpful logging. I think we should check if the user has any of the relationships already on their library card before making any more relationships so we don't end up in a halfway-state where some of the DAAs in the list get added and some don't.

Comment (fboulnois) [reply src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java:298]: Let's discuss what the expected behavior should be.

Comment (fboulnois) [reply src/main/java/org/broadinstitute/consent/http/resources/DaaResource.java:298]: This should fix the issue: https://github.com/DataBiosphere/consent/pull/2308/commits/6a25bf822186af72d85568e30d9fa56d5d90a214

**Review by aarohinadkarni — APPROVED** [review]
thanks for the changes! 👍

**Pull request merged**